help.go

```diff
@@ -276,3 +292,27 @@ func formatFlag(haveShort bool, flag *Flag) string {
 	}
 	return flagString
 }
+
+// commandTree creates a tree with the given node name as root and its children's arguments and sub commands as leafs.
+func commandTree(node *Node, prefix string) (rows [][2]string) {
+	var nodeName string
+	switch node.Type {
+	case CommandNode:
+		nodeName += prefix + node.Name
+	case ArgumentNode:
+		nodeName += prefix + "<" + node.Name + ">"
+	}
+	rows = append(rows, [2]string{nodeName, node.Help})
+	if prefix == "" {
+		prefix = "|- "
```

Comment:
Did you try this with just spaces? I don't love the "|-" look, aesthetically.

Reply:
I could also make it optional. If Compact is enabled, using only blank spaces, otherwise using the '|-' prefixes. Alternatively, I could transform the Views to an enum field. Or maybe you have another idea how it could be drawed more nicely?

How about that:
```bash
Command Tree:
  version               display the app version

  help                  show help for a command
  | [<command> ...]     the commands for which to show help

  build                 build Docker tags from the given tag vectors
  | from                determine the source of the tag vectors
  | | stdin             read the tag vectors from the standard input
  | | file              read the tag vectors from a Dockerfile
  | | | <file>          the Dockerfile containing the vectors as FROM instructions
  | | <param>           pass the tag vectors as arguments

  tag                   tag the given source image with the Docker tags from the given tag vectors
  | <source-tag>        the source tag of the image that should receive the generated tags
  | | from              determine the source of the tag vectors
  | | | stdin           read the tag vectors from the standard input
  | | | file            read the tag vectors from a Dockerfile
  | | | | <file>        the Dockerfile containing the vectors as FROM instructions
  | | | <param>         pass the tag vectors as arguments
  | | to                set a target repository
  | | | <repository>    the Docker Hub repository of the root tag vector in the format 'organization/repository'
  | | | | from          determine the source of the tag vectors
  | | | | | stdin       read the tag vectors from the standard input
  | | | | | file        read the tag vectors from a Dockerfile
  | | | | | | <file>    the Dockerfile containing the vectors as FROM instructions
  | | | | | <param>     pass the tag vectors as arguments
```

Or that one:
```bash
Command Tree:
  version               display the app version

  help                  show help for a command
  - [<command> ...]     the commands for which to show help

  build                 build Docker tags from the given tag vectors
  - from                determine the source of the tag vectors
    - stdin             read the tag vectors from the standard input
    - file              read the tag vectors from a Dockerfile
      - <file>          the Dockerfile containing the vectors as FROM instructions
    - <param>           pass the tag vectors as arguments

  tag                   tag the given source image with the Docker tags from the given tag vectors
  - <source-tag>        the source tag of the image that should receive the generated tags
    - from              determine the source of the tag vectors
      - stdin           read the tag vectors from the standard input
      - file            read the tag vectors from a Dockerfile
        - <file>        the Dockerfile containing the vectors as FROM instructions
      - <param>         pass the tag vectors as arguments
    - to                set a target repository
      - <repository>    the Docker Hub repository of the root tag vector in the format 'organization/repository'
        - from          determine the source of the tag vectors
          - stdin       read the tag vectors from the standard input
          - file        read the tag vectors from a Dockerfile
            - <file>    the Dockerfile containing the vectors as FROM instructions
          - <param>     pass the tag vectors as arguments
```

Or by giving help option that takes a function that just does the prefixing and providing some exported views as templates.

```golang
  // Tree writes command chains in a tree structure instead of listing them separately.
  Tree bool

  // TreePrefix modulates the given prefix for the next layer in the tree view.
  // The following templates can be used by default: SpaceTree....
  TreePrefix func(prefix string) string
}

func SpaceTree(prefix string) string {}
```

Reply:
I like that last option with a function!